File: common/src/main/kotlin/io/edanni/rndl/common/domain/entity/Entry.kt
package io.edanni.rndl.common.domain.entity

import com.vividsolutions.jts.geom.Geometry
import org.threeten.bp.LocalDateTime
import org.threeten.bp.OffsetDateTime
import java.math.BigDecimal

data class Entry(
        override val id: Long? = null,
        override val createdAt: OffsetDateTime? = null,
        override val updatedAt: OffsetDateTime? = null,
        val tripId: Long? = null,
        val deviceTime: LocalDateTime? = null,
        val coordinates: Geometry? = null,
        val gpsSpeed: BigDecimal? = null,
        val gpsAltitude: BigDecimal? = null,
        val rpm: BigDecimal? = null,
        val economy: BigDecimal? = null,
        val speed: Int? = null,
        val throttle: BigDecimal? = null,
        val instantEconomy: BigDecimal? = null,
        val fuelFlow: BigDecimal? = null,
        val fuelUsed: BigDecimal? = null
) : Entity

File: webapp/src/api/entities/vehicle/Entry.ts
import Entity from '../Entity';
import { Point } from 'geojson';
import Trip from './Trip';

export default interface Entry extends Entity {
    tripId?: number;
    deviceTime?: string;
    coordinates?: Point;
    gpsSpeed?: number;
    gpsAltitude?: number;
    rpm?: number;
    economy?: number;
    speed?: number;
    throttle?: number;
    instantEconomy?: number;
    fuelFlow?: number;
    fuelUsed?: number;
    trip?: Trip;
}